lamindb/models/artifact.py

```diff
@@ -523,45 +523,59 @@ def log_storage_hint(
     logger.hint(hint)
 
 
-def data_is_anndata(data: AnnData | UPathStr) -> bool:
-    if isinstance(data, AnnData):
+def data_is_scversedatastructure(
+    data: ScverseDataStructures | UPathStr,
+    expected_ds: Literal["AnnData", "MuData", "SpatialData"] | None = None,
+) -> bool:
+    """Determine whether a specific in-memory object or a UPathstr is any or a specific scverse data structure."""
+    file_suffix = None
+    if expected_ds == "AnnData":
+        file_suffix = ".h5ad"
+    elif expected_ds == "MuData":
+        file_suffix = ".h5mu"
+    # SpatialData does not have a unique suffix but `.zarr`
+
+    if expected_ds is None:
+        return any(
+            hasattr(data, "__class__") and data.__class__.__name__ == cl_name
+            for cl_name in ["AnnData", "MuData", "SpatialData"]
+        )
+    elif hasattr(data, "__class__") and data.__class__.__name__ == expected_ds:
         return True
+
+    data_type = expected_ds.lower()
     if isinstance(data, (str, Path, UPath)):
         data_path = UPath(data)
-        if ".h5ad" in data_path.suffixes:  # ".h5ad.gz" is a valid suffix
+
+        if file_suffix and f"{file_suffix}" in data_path.suffixes:
```

Comment:
Why not `file_suffix in data_path.suffixes`?

Reply:
Whopsie, not sure what happened here. Will fix.